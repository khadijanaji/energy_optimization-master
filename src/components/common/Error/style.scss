@import "../../../style/colors.scss";
.reporting-error-message {
  display: inline-block;
  width: 90%;
  text-align: center;
  padding-top: 3%;
  padding-bottom: 3%;
  padding-left: 5%;
  padding-right: 5%;
  font-family: 'Roboto';
  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  text-align: center;
  background-color: $text-color-danger; 
  color: $header-text-color;
}