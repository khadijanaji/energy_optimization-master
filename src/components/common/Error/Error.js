// @flow

import React from "react";
import "./style.scss";

type ErrorProps = {
    text?: string,
    refreshFn?: Function
}

const Error = ({text = "Erreur", refreshFn}: ErrorProps) => (
  <div
    className="reporting-error-message"
    onClick={refreshFn}
  >{text}
  </div>
);

export default Error;
