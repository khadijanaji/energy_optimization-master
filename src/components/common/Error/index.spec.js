/* globals describe it expect */

import {shallow} from "enzyme";
import React from "react";

import Error from "./";

describe("Component", () => {
  describe("Error Render", () => {
    it("Without Props: Should match snapshot", () => {
      const wrapper = shallow(<Error />);
      expect(wrapper).toMatchSnapshot();
    });
    it("With Props: Should match snapshot", () => {
      const wrapper = shallow(<Error text={"Props Erreur"} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
