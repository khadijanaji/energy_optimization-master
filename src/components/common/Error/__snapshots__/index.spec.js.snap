// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Error Render With Props: Should match snapshot 1`] = `
<div
  className="reporting-error-message"
>
  Props Erreur
</div>
`;

exports[`Component Error Render Without Props: Should match snapshot 1`] = `
<div
  className="reporting-error-message"
>
  Erreur
</div>
`;
